lib/shoryuken/manager.rb

```diff
@@ -154,6 +153,13 @@ def real_thread(proxy_id, thr)
 
     private
 
+    def dispatch_later
+      @_dispatch_timer ||= after(1) do
+        @_dispatch_timer = nil
+        dispatch
+      end
+    end
```

Comment:
@atyndall BTW I was able to cherry-pick your commit, to give you the credit for the patch 🍻